cpp/src/arrow/compute/registry.cc

```diff
@@ -115,20 +210,42 @@ std::unique_ptr<FunctionRegistry> FunctionRegistry::Make() {
   return std::unique_ptr<FunctionRegistry>(new FunctionRegistry());
 }
 
-FunctionRegistry::FunctionRegistry() { impl_.reset(new FunctionRegistryImpl()); }
+std::unique_ptr<FunctionRegistry> FunctionRegistry::Make(FunctionRegistry* parent) {
+  return std::unique_ptr<FunctionRegistry>(new FunctionRegistry(
```

Comment:
Instead, can we use `arrow::internal::make_unique`?

Reply:
I'll try this.

Reply:
This fails to compile because `explicit FunctionRegistry(FunctionRegistryImpl* impl)` is private (see compiler output below). Looks like the existing code works because it is all within the scope of `FunctionRegistry` whereas using `make_unique` introduces another scope.
```
In file included from /mnt/user1/tscontract/github/rtpsw/arrow/cpp/src/arrow/compute/registry.cc:19:
/mnt/user1/tscontract/github/rtpsw/arrow/cpp/src/arrow/util/make_unique.h: In instantiation of ‘typename std::enable_if<(! std::is_array< <template-parameter-1-1> >::value), std::unique_ptr<T> >::type arrow::internal::make_unique(A&& ...) [with T = arrow::compute::FunctionRegistry; A = {arrow::compute::FunctionRegistry::FunctionRegistryImpl*}; typename std::enable_if<(! std::is_array< <template-parameter-1-1> >::value), std::unique_ptr<T> >::type = std::unique_ptr<arrow::compute::FunctionRegistry>]’:
/mnt/user1/tscontract/github/rtpsw/arrow/cpp/src/arrow/compute/registry.cc:216:70:   required from here
/mnt/user1/tscontract/github/rtpsw/arrow/cpp/src/arrow/util/make_unique.h:30:29: error: ‘arrow::compute::FunctionRegistry::FunctionRegistry(arrow::compute::FunctionRegistry::FunctionRegistryImpl*)’ is private within this context
   30 |   return std::unique_ptr<T>(new T(std::forward<A>(args)...));
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/user1/tscontract/github/rtpsw/arrow/cpp/src/arrow/compute/registry.cc:221:1: note: declared private here
  221 | FunctionRegistry::FunctionRegistry(FunctionRegistryImpl* impl) { impl_.reset(impl); }
      | ^~~~~~~~~~~~~~~~
```

Reply:
I see. May be we can stick to what we had. cc @lidavidm